cli/azd/cmd/middleware/middleware.go

```diff
@@ -48,15 +48,17 @@ type NextFn func(ctx context.Context) (*actions.ActionResult, error)
 // Middleware runner stores middleware registrations and orchestrates the
 // invocation of middleware components and actions.
 type MiddlewareRunner struct {
-	chain     []string
-	container *ioc.NestedContainer
+	chain       []string
+	container   *ioc.NestedContainer
+	actionCache map[actions.Action]*actions.ActionResult
```

Comment:
Similar question about caching here -- do we need it, especially for this particular change? Or can this be a follow-up?

Reply:
This depends.  The use case is as follows:

`azd deploy` => requires package
`azd provision` => requires package
`azd up` => Calls both `provision` & `deploy`

We don't want to re-package multiple times on the `up` flow, so we cache the first response and reuse when it is available.